File: 2109_AddingSpacesToAString.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        int lengthOfS=s.length(), 
            lengthOfSpaces=spaces.length, 
            requiredLength=lengthOfS+lengthOfSpaces,
            currentSpaceIndex=0, currentTrvIndex=0;
        StringBuilder result=new StringBuilder();
        for(int i=0; i<requiredLength; ++i){
            result.append(' ');
        }
        for(int i=0; i<lengthOfS; ++i){
            char currentChar=s.charAt(i);
            if(currentSpaceIndex<lengthOfSpaces&&i==spaces[currentSpaceIndex]){
                currentTrvIndex++;
                result.setCharAt(currentTrvIndex, currentChar);
                currentSpaceIndex++;
                currentTrvIndex++;
            } else{
                result.setCharAt(currentTrvIndex, currentChar);
                currentTrvIndex++;
            }
        }
        return String.valueOf(result);
    }
}

File: 2/2109/2109_AddingSpacesToAString.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        lengthOfS=len(s)
        lengthOfS = len(s)
        lengthOfSpaces = len(spaces)
        result = []
        currentSpaceIndex = 0
        currentTraversingIndex = 0
        
        for i in range(lengthOfS):
            if currentSpaceIndex < lengthOfSpaces and i == spaces[currentSpaceIndex]:
                result.append(' ')
                currentSpaceIndex += 1
            result.append(s[i])
        
        return ''.join(result)
